### PR TITLE
Install Appnet Agent from AL repo; disable userland-proxy for docker

### PR DESCRIPTION
**Pull request by mythri-garaga**
### Summary

This PR includes changes to 
1. Install Appnet agent required during AMI build for al2. Appnet Agent rpm is not available for al2022 yet.
2. Disable userland-proxy for docker to avoid using up host ephemeral ports when client and server are colocated on host for SC bridge mode.

### Testing

Tested the changes manually by building a new AMI and confirmed that
1. Agent successfully loaded the Appnet tarball for the expected filepath
2. `userland-proxy` config disabled in /etc/docker/daemon.json
3.  Was able to run SC task in bridge mode successfully and see that Appnet agent and Appnet relay containers were running.

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: al2.pkr.hcl [hunk 1]
```diff
@@ -174,6 +174,10 @@ build {
     script = "scripts/enable-services.sh"
   }
 
+  provisioner "shell" {
```
Comment (ubhattacharjya) [inline al2.pkr.hcl:177]: I see this only for al2. Is it not needed for al2022?

Comment (mythri-garaga) [reply al2.pkr.hcl:177]: We don't have rpm available for al2022 yet. I will update the description as well, thanks for calling it out.

**Review by yinyic — COMMENTED** [review]
Let's verify with a test AMI build and run an SC task on the AMI, if we haven't.

File: scripts/install-docker.sh [hunk 2]
```diff
@@ -12,3 +12,15 @@ if command -v amazon-linux-extras; then
 fi
 
 sudo yum install -y "docker-$DOCKER_VERSION" "containerd-$CONTAINERD_VERSION"
+
+# disable userland-proxy for docker
```
Comment (chienhanlin) [inline scripts/install-docker.sh:16]: Is this change applicable to AL2022? `install-docker.sh` is used in the AL2022 recipe [here](https://github.com/aws/amazon-ecs-ami/blob/main/al2022.pkr.hcl#L86) as well.

Comment (mythri-garaga) [reply scripts/install-docker.sh:16]: Yes and @yinyic ran regression testing for AL2022 and tests succeeded(with couple of expected failures).

**Review by ubhattacharjya — APPROVED** [review]

**Review by chienhanlin — APPROVED** [review]

**Pull request merged**